src/mixing_fab_tests.cc

```diff
@@ -0,0 +1,247 @@
+  
+  void MixingFabTest::TestBuffers(MixingFab* fac,
+                                  std::vector<double> in_inv,
+                                  double out_inv){
+    for (int i = 0; i<in_com.size(); i++) {
+      EXPECT_EQ(in_inv[i], fac->streambufs[in_com[i]].quantity());
+    }
+    EXPECT_EQ(out_inv, fac->output.quantity());
+  }
+  
+  void MixingFabTest::TestOutputComposition(MixingFab* fac, cyclus::CompMap v){
+    
+    cyclus::toolkit::ResBuf<cyclus::Material>* buffer = &fac->output;
+    Material::Ptr final_mat = cyclus::ResCast<Material>(buffer->PopBack());
+    cyclus::CompMap final_comp = final_mat->comp()->mass();
+    Normalize(&v,1);
+    Normalize(&final_comp,1);
+    
+    cyclus::CompMap sum_v = Add(v, final_comp);
+    
+    CompMap::iterator it;
+    for(it = sum_v.begin(); it != sum_v.end(); it ++){
+      EXPECT_DOUBLE_EQ(final_comp[it->first],v[it->first]);
+    }
+
+    
+  }
+  
+  
+  
+  
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+  TEST_F(MixingFabTest, InitialState) {
+    // Test things about the initial state of the facility here
+    TestInitState(mf_facility_);
+  }
+
+  
+  
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+  // multiple input streams can be correctly requested and used as
+  //  material inventory.
+  TEST_F(MixingFabTest, MultipleFissStreams) {
+    std::string config =
+    "<in_commods> <val>stream1</val> <val>stream2</val> <val>stream3</val> </in_commods>"
+    "<in_buf_size> <val>2.5</val> <val>3</val> <val>5</val></in_buf_size>"
+    "<mixing_frac> <val>0.8</val> <val>0.15</val> <val>0.05</val></mixing_frac>"
+    ""
+    "<out_commod>dummyout</out_commod>"
+    "<outputbuf_size>0</outputbuf_size>"
+    "<throughput>0</throughput>"
+    ;
+    
+    int simdur = 1;
+    cyclus::MockSim sim(cyclus::AgentSpec(":cycamore:MixingFab"), config, simdur);
+    sim.AddSource("stream1").recipe("unatstream").capacity(1).Finalize();
+    sim.AddSource("stream2").recipe("uoxstream").capacity(1).Finalize();
+    sim.AddSource("stream3").recipe("pustream").capacity(1).Finalize();
+    sim.AddRecipe("unatstream", c_natu());
+    sim.AddRecipe("uoxstream", c_pustream());
+    sim.AddRecipe("pustream", c_uox());
+    int id = sim.Run();
+    
+    QueryResult qr = sim.db().Query("Transactions", NULL);
+    EXPECT_EQ(3, qr.rows.size());
+    
+    std::vector<Cond> conds;
+    conds.push_back(Cond("Commodity", "==", std::string("stream1")));
+    qr = sim.db().Query("Transactions", &conds);
+    EXPECT_EQ(1, qr.rows.size());
+    
+    conds[0] = Cond("Commodity", "==", std::string("stream2"));
+    qr = sim.db().Query("Transactions", &conds);
+    EXPECT_EQ(1, qr.rows.size());
+    
+    conds[0] = Cond("Commodity", "==", std::string("stream3"));
+    qr = sim.db().Query("Transactions", &conds);
+    EXPECT_EQ(1, qr.rows.size());
+  }
+  
+  
+  TEST_F(MixingFabTest, MixingComposition) {
+    throughput = 1e200;
+    SetUpMixingFab();
```

Comment:
delete this call - setup/teardown functions are already automatically run for before/after each test.